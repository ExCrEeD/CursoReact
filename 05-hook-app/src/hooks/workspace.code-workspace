{
	"folders": [
		{
			"path": "..\\..\\..\\07-heroes-app"
		}
	],
	"settings": {}
}