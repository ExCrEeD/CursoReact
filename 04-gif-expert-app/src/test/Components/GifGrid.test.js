import { GifGrid } from "../../Components/GifGrid";
import {shallow} from 'enzyme';
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('pruebas GifGrid', () => {
  
    let category = 'HunterXHunter';
    test('debe de renderizar el componente', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });
        let wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('debe de mostar items cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs =[{
            id:'ABC',
            url:'https://adksadl/asdas.jpg',
            title:'Cuaqlueir cosa'
        },
        {
            id:'123',
            url:'https://adksadl/asdas.jpg',
            title:'prueab 2'
        }]

        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        });
        let wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
});
