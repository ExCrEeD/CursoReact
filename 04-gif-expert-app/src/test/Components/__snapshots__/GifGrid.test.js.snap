// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas GifGrid debe de mostar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    HunterXHunter
  </h3>
  <div
    className="cardGrid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cuaqlueir cosa"
      url="https://adksadl/asdas.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="prueab 2"
      url="https://adksadl/asdas.jpg"
    />
  </div>
</Fragment>
`;

exports[`pruebas GifGrid debe de renderizar el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    HunterXHunter
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading
  </p>
  <div
    className="cardGrid"
  />
</Fragment>
`;
